submissions/7SergSmith7/Friends-app/app.js

```diff
@@ -0,0 +1,142 @@
+
+    .catch((error) => {
+      console.error("Error:", error);
+    });
+}
+
+function renderContactTemplate(contact) {
+  const { gender, name, location, dob, phone, picture } = contact;
+  return `<div class="contact__item" data-name="${name.first}">
+            <img class="contact__image" src=${picture.large} alt=${name.first}>
+            <h3 class="contact__title">${name.first} ${name.last}</h3>
+            <p class="contact__phone">${phone}</p>   
+            <div class="contact__additional">
+              <div class="contact__additional__gender">
+                <h4 class="additional__gender__title">sex</h4>
+                <p class="additional__gender_item">${gender}</p>
+              </div>
+              <div class="contact__additional__age">
+                <h4 class="additional__age__title">age</h4>
+                <p class="additional__age_item">${dob.age}</p>
+              </div>
+              <div class="contact__additional__location">
+              <h4 class="additional__location__title">location</h4>   
+              <p class="additional__location_item">${location.country}</p>     
+              </div>
+            </div>
+          </div>`;
+}
+
+optionsMenu.addEventListener("click", onOptionsClick);
+searchInput.addEventListener("change", onChangeSearch);
+
+function onChangeSearch() {
+  contacts = contacts.filter((element) => {
+    return element.name.first
+      .toLowerCase()
+      .includes(searchInput.value.toLowerCase());
+  });
+  renderContacts(contacts);
+}
+
+function renderContacts(contacts) {
+  contactsField.innerHTML = contacts
+    .map((contact) => renderContactTemplate(contact))
+    .join("");
+}
+
+function onOptionsClick(e) {
+  const selectedOption = e.target.classList.value;
+
+if(selectedOption=="options__blocks__reset-btn"){
+  resetOptions();
+}
+
+
+if(selectedOption.includes("name-sort")){
+  sortName(contacts);
+  if(selectedOption.includes("a-z"))
+  contacts = contacts.reverse();
+    
+}
+
+if(selectedOption.includes("age-sort")){
+  sortAge(contacts);
+  if(selectedOption.includes("less"))
+  contacts = contacts.reverse();
+   
+}
+
+if(selectedOption.includes("gender-filter")){
+  if(selectedOption.includes("man"))
+  filter = "male";
+  else filter = "female";
+   
+}
+renderContacts(filterByGenderIfYouHaveAGender(contacts, filter));
+
+
+ }
+
+function resetOptions() {
+  resetToDefaultContacts();
+  filter = "";
+  searchInput.value = "";
+  
+}
+
+function sortAge(contact) {
+  contacts = contact.sort((a, b) => a.dob.age - b.dob.age);
+}
+
+function sortName(contact) {
+  contacts = contact.sort(function (a, b) {
+    if (a.name.first > b.name.first) {
+      return -1;
+    }
+    if (a.name.first < b.name.first) {
+      return 1;
+    }
+    return 0;
+  });
+}
+function filterGender(contacts, gender) {
+  return contacts.filter((element) => element.gender == gender);
+}
+
+function filterByGenderIfYouHaveAGender(contacts, gender) {
```

Comment:
`filterByGenderIfYouHaveAGenderAndIfYouDon'tHaveGenderReturnContacts`

Just call it `filterByGender` and that's all ;)